### PR TITLE
Plasma cutter tweaks

### PR DESCRIPTION
**Pull request by Kharkov72**
Really though why did it say nowhere on it that it reloads by slapping plasma into it FUCK
also I'm going to throw in the same fix as #23286 because this has the same problem

:cl:
 * rscadd: Plasma Cutter description now says how to load it, and displays remaining ammo
 * bugfix: Plasma Cutters can now hit people lying down, and the conversion kit will no longer eat multiple accelerators
 * rscadd: Plasma Cutters can now be loaded with plasma ore
 * bugfix: Plasma Cutters now go through the normal weapon checks (Looking at you, mommis)

### CODE REVIEW

File: code/modules/mining/mine_items.dm [hunk 1]
```diff
@@ -270,7 +270,7 @@ proc/move_mining_shuttle()
 
 /obj/item/weapon/pickaxe/plasmacutter/accelerator
 	name = "plasma cutter"
-	desc = "A rock cutter that fires bolts of hot plasma. You could use it to cut limbs off of xenos! Or, you know, mine stuff."
```
Comment (MadmanMartian) [inline code/modules/mining/mine_items.dm:273]: "A rock cutter that utilizes plasma to create sharp beams of energy. The cutting power is enough to cut through rocks and xenos with ease. It requires solid plasma to function.

Comment (ShiftyRail) [reply code/modules/mining/mine_items.dm:273]: I preferred the little sillyness of the old one.

File: code/modules/mining/mine_items.dm [hunk 2]
```diff
@@ -287,7 +287,7 @@ proc/move_mining_shuttle()
 		var/turf/target = get_turf(A)
```
Comment (MadmanMartian) [inline code/modules/mining/mine_items.dm]: There's a generic fire proc that covers all this bullshit, why is it not being used?

Comment (Kharkov72) [reply code/modules/mining/mine_items.dm]: Show me the way sensei

Comment (Kharkov72) [reply code/modules/mining/mine_items.dm]: the deed is done

**Comment by utilitas-** [comment]
The plasma cutter should accept raw plasma ore as well. Only guzzling refined plasma makes it strictly inferior to a plain protokinetic accelerator, which doesn't need to waste resources to recharge. Furthermore, it should have a check so it doesn't eat an entire stack of plasma, only one sheet.

**Comment by Kharkov72** [comment]
Tested, can now be loaded with ore.

File: code/modules/mining/mine_items.dm [hunk 3]
```diff
@@ -290,16 +290,25 @@ proc/move_mining_shuttle()
...
-				to_chat(user, "<span class='notice'>You load [src].</span>")
-			else
-				to_chat(user, "<span class='notice'>[src] is already loaded.</span>")
+	if(proximity_flag && istype(target, /obj/item/stack/ore/plasma))
+		var/obj/item/stack/ore/plasma/A = target
+		if(current_ammo < max_ammo)
+			var/loading_ammo = min(max_ammo - current_ammo, A.amount)
+			A.use(loading_ammo)
+			current_ammo += loading_ammo
+			to_chat(user, "<span class='notice'>You load [src].</span>")
```
Comment (MadmanMartian) [inline code/modules/mining/mine_items.dm]: You load \the [src]

Comment (Kharkov72) [reply code/modules/mining/mine_items.dm]: sir yes sir

**Comment by utilitas-** [comment]
Another thing: You can add multiple proto-kinetic accelerators to the conversion kit. There should be a check to require only one accelerator.

**Comment by Kharkov72** [comment]
jesus christ why did it only have a check to load and not to check if it was loaded

**Comment by MadmanMartian** [comment]
can we have an examine override for it to show how many shots it has left?

**Comment by Kharkov72** [comment]
Untitled
sure chief

**Pull request merged**